### ISSUE TITLE
Getting started guide ScyllaCluster resource fails with Init:InvalidImageName

### ISSUE THREAD

**Issue by eighty4** [scylladb/scylla-operator#1441]
I followed the instructions in the getting started guide, first installing cert-manager.yml and then the operator and cluster CRDs. When applying the provided sample ScyllaCluster resource I get an `Init:InvalidImageName` error.

`kubectl describe pod/my-scylladb-us-east-1-us-east-1a-0` shows this error event for startup:

```
Events:
  Type     Reason            Age                   From               Message
  ----     ------            ----                  ----               -------
  Warning  FailedScheduling  14m                   default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
  Normal   Scheduled         14m                   default-scheduler  Successfully assigned default/my-scylladb-us-east-1-us-east-1a-0 to minikube
  Warning  FailedMount       14m (x5 over 14m)     kubelet            MountVolume.SetUp failed for volume "scylladb-serving-certs" : secret "my-scylladb-local-serving-certs" not found
  Warning  Failed            13m (x10 over 14m)    kubelet            Error: InvalidImageName
  Warning  InspectFailed     4m43s (x48 over 14m)  kubelet            Failed to apply default image tag "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": couldn't parse image reference "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": invalid reference format
```

1. Apply cert-manager.yml, operator.yml, and cluster.yml
2. `kubectl describe pod/my-scylladb-us-east-1-us-east-1a-0`
3. See error

**Expected behavior**
Node starts up per getting started guide.

**Logs**

```
Name:             my-scylladb-us-east-1-us-east-1a-0
Namespace:        default
Priority:         0
Service Account:  my-scylladb-member
Node:             minikube/192.168.49.2
Start Time:       Mon, 02 Oct 2023 20:34:33 -0500
Labels:           app=scylla
                  app.kubernetes.io/managed-by=scylla-operator
                  app.kubernetes.io/name=scylla
                  controller-revision-hash=my-scylladb-us-east-1-us-east-1a-7474c788cb
                  scylla/cluster=my-scylladb
                  scylla/datacenter=us-east-1
                  scylla/rack=us-east-1a
                  scylla/rack-ordinal=0
                  scylla/scylla-version=5.2.7
                  statefulset.kubernetes.io/pod-name=my-scylladb-us-east-1-us-east-1a-0
Annotations:      prometheus.io/port: 9180
                  prometheus.io/scrape: true
Status:           Pending
IP:               10.244.0.15
IPs:
  IP:           10.244.0.15
Controlled By:  StatefulSet/my-scylladb-us-east-1-us-east-1a
Init Containers:
  sidecar-injection:
    Container ID:
    Image:         docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c
      cp -a /usr/bin/scylla-operator /mnt/shared
    State:          Waiting
      Reason:       InvalidImageName
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     10m
      memory:  50Mi
    Requests:
      cpu:        10m
      memory:     50Mi
    Environment:  <none>
    Mounts:
      /mnt/shared from shared (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-hpzhl (ro)
Containers:
  scylla:
    Container ID:
    Image:         docker.io/scylladb/scylla:5.2.7
    Image ID:
    Ports:         7000/TCP, 7001/TCP, 7199/TCP, 9180/TCP, 9100/TCP, 9042/TCP, 9142/TCP, 9160/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
    Command:
      /mnt/shared/scylla-operator
      sidecar
      --feature-gates=AllAlpha=false,AllBeta=false,AutomaticTLSCertificates=true
      --nodes-broadcast-address-type=ServiceClusterIP
      --clients-broadcast-address-type=ServiceClusterIP
      --service-name=$(SERVICE_NAME)
      --cpu-count=$(CPU_COUNT)
      --loglevel=2
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     1
      memory:  1Gi
    Requests:
      cpu:      1
      memory:   1Gi
    Liveness:   http-get http://:8080/healthz delay=0s timeout=10s period=10s #success=1 #failure=12
    Readiness:  http-get http://:8080/readyz delay=0s timeout=30s period=10s #success=1 #failure=1
    Startup:    http-get http://:8080/healthz delay=0s timeout=30s period=10s #success=1 #failure=40
    Environment:
      SERVICE_NAME:  my-scylladb-us-east-1-us-east-1a-0 (v1:metadata.name)
      CPU_COUNT:     1 (limits.cpu)
    Mounts:
      /mnt/scylla-client-config from scylla-client-config-volume (ro)
      /mnt/scylla-config from scylla-config-volume (ro)
      /mnt/shared from shared (ro)
      /tmp/coredumps from coredumpfs (rw)
      /var/lib/scylla from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-hpzhl (ro)
      /var/run/secrets/scylla-operator.scylladb.com/scylladb/client-ca from scylladb-client-ca (ro)
      /var/run/secrets/scylla-operator.scylladb.com/scylladb/serving-certs from scylladb-serving-certs (ro)
      /var/run/secrets/scylla-operator.scylladb.com/scylladb/user-admin from scylladb-user-admin (ro)
  scylla-manager-agent:
    Container ID:
    Image:         docker.io/scylladb/scylla-manager-agent:3.1.2
    Image ID:
    Port:          10001/TCP
    Host Port:     0/TCP
    Args:
      -c
      /etc/scylla-manager-agent/scylla-manager-agent.yaml
      -c
      /mnt/scylla-agent-config/scylla-manager-agent.yaml
      -c
      /mnt/scylla-agent-config/auth-token.yaml
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Requests:
      cpu:        50m
      memory:     10M
    Environment:  <none>
    Mounts:
      /mnt/scylla-agent-config/auth-token.yaml from scylla-agent-auth-token-volume (ro,path="auth-token.yaml")
      /mnt/scylla-agent-config/scylla-manager-agent.yaml from scylla-agent-config-volume (ro,path="scylla-manager-agent.yaml")
      /var/lib/scylla from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-hpzhl (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  data-my-scylladb-us-east-1-us-east-1a-0
    ReadOnly:   false
  shared:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  scylla-config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      scylla-config
    Optional:  true
  scylla-agent-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  scylla-agent-config
    Optional:    true
  scylla-client-config-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  scylla-client-config-secret
    Optional:    true
  scylla-agent-auth-token-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  my-scylladb-auth-token
    Optional:    false
  scylladb-serving-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  my-scylladb-local-serving-certs
    Optional:    false
  scylladb-client-ca:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  my-scylladb-local-client-ca
    Optional:    false
  scylladb-user-admin:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  my-scylladb-local-user-admin
    Optional:    false
  coredumpfs:
    Type:          HostPath (bare host directory volume)
    Path:          /tmp/coredumps
    HostPathType:
  kube-api-access-hpzhl:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason            Age                   From               Message
  ----     ------            ----                  ----               -------
  Warning  FailedScheduling  14m                   default-scheduler  0/1 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
  Normal   Scheduled         14m                   default-scheduler  Successfully assigned default/my-scylladb-us-east-1-us-east-1a-0 to minikube
  Warning  FailedMount       14m (x5 over 14m)     kubelet            MountVolume.SetUp failed for volume "scylladb-serving-certs" : secret "my-scylladb-local-serving-certs" not found
  Warning  Failed            13m (x10 over 14m)    kubelet            Error: InvalidImageName
  Warning  InspectFailed     4m43s (x48 over 14m)  kubelet            Failed to apply default image tag "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": couldn't parse image reference "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": invalid reference format
```

**Environment:**
 - Platform: minikube
 - Minikube version: 1.31.2
 - Kubernetes version: 1.27.4
 - Scylla version: 5.2.7
 - Scylla-operator version: e.g.: master

**Comment by tnozicka** [comment]
> Warning  InspectFailed     4m43s (x48 over 14m)  kubelet            Failed to apply default image tag "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": couldn't parse image reference "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": invalid reference format

I suspect there is something wrong with your Kube setup / container runtime.
Are other apps not using an explicit repo working fine?
Does prefixing the repo with `docker.io/` help?


> Kubernetes version: 1.31.2

this seem to be a release from a future

**Comment by eighty4** [comment]
I just opened this ticket to replace the minikube version with the kubernetes version. It's 1.27.4.

**Comment by rzetelskik** [comment]
@tnozicka is docker-pullable even a valid transport?
```console
$ skopeo inspect docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6
FATA[0000] Error parsing image name "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6": Invalid image name "docker-pullable://scylladb/scylla-operator@sha256:96dfee085c076f74cf5f2b128266a0dfd86e4008c4c97783414d74a310de99b6", unknown transport "docker-pullable"
```

**Comment by eighty4** [comment]
I couldn't find a reference via the google or github search on this repo, but it's prevalent all over other github repos

**Comment by zimnx** [comment]
> Apply cert-manager.yml, operator.yml, and **cluster.yml**

None of our examples creates a cluster called `my-scylladb`. Please provide what ScyllaCluster you actually applied.

**Comment by tnozicka** [comment]
can you provide logs from the scylla-operator pod?

**Comment by eighty4** [comment]
I dropped the operator in favor of running individual nodes. I'm not sure what problems the operator solves, but I feel like I would learn more running Scylla directly. I ran a couple 20+ node clusters of Cassandra and I'm excited to see how Scylla is in comparison. Running Scylla via 10000 lines of yaml and unknown additional binaries orchestrating in k8s scares me more than manually operating a couple nodes.

If I revisit this weekend I'll paste the operator logs.

</monlogue>
